workflow/executor/docker/docker.go

```diff
@@ -162,8 +164,30 @@ func (d *DockerExecutor) GetExitCode(ctx context.Context, containerID string) (s
 	return exitCode, nil
 }
 
-func (d *DockerExecutor) WaitInit() error {
-	return nil
+func (d *DockerExecutor) WaitInit(ctx context.Context) (string, error) {
+	for {
+		select {
+		case <-ctx.Done():
+			return "", ctx.Err()
+		default:
+			output, err := common.RunCommand(
+				"docker",
+				"ps",
+				"--all",      // container could have already exited
+				"--no-trunc", // long container ID please
```

Comment:
@Random-Liu I hope you don't mind me cold-mentioning you - but I wonder if this is a reasonable way to determine the pod and container of a docker process?

Reply:
thank you in advance!